modules/service/src/main/scala/lucuma/odb/service/ObservationWorkflowService.scala

```diff
@@ -0,0 +1,461 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.service
+
+import cats.data.NonEmptyList
+import cats.data.OptionT
+import cats.effect.Concurrent
+import cats.implicits.*
+import grackle.Result
+import grackle.ResultT
+import lucuma.core.enums.CalibrationRole
+import lucuma.core.enums.ConfigurationRequestStatus
+import lucuma.core.enums.Instrument
+import lucuma.core.enums.ObsActiveStatus
+import lucuma.core.enums.ObservationValidationCode
+import lucuma.core.enums.ScienceBand
+import lucuma.core.enums.Site
+import lucuma.core.math.Coordinates
+import lucuma.core.math.Declination
+import lucuma.core.math.RightAscension
+import lucuma.core.model.CallForProposals
+import lucuma.core.model.ConfigurationRequest
+import lucuma.core.model.ObjectTracking
+import lucuma.core.model.Observation
+import lucuma.core.model.ObservationValidation
+import lucuma.core.model.ObservingNight
+import lucuma.core.model.Program
+import lucuma.core.model.StandardRole.*
+import lucuma.core.model.Target
+import lucuma.core.syntax.string.*
+import lucuma.core.util.DateInterval
+import lucuma.core.util.Enumerated
+import lucuma.itc.client.ItcClient
+import lucuma.odb.data.ObservationExecutionState
+import lucuma.odb.data.ObservationValidationMap
+import lucuma.odb.data.ObservationWorkflow
+import lucuma.odb.data.ObservationWorkflowState
+import lucuma.odb.data.Tag
+import lucuma.odb.graphql.enums.Enums
+import lucuma.odb.logic.TimeEstimateCalculatorImplementation
+import lucuma.odb.sequence.data.GeneratorParams
+import lucuma.odb.sequence.data.MissingParamSet
+import lucuma.odb.sequence.util.CommitHash
+import lucuma.odb.service.GeneratorParamsService.Error as GenParamsError
+import lucuma.odb.syntax.instrument.*
+import lucuma.odb.util.Codecs.*
+import natchez.Trace
+import skunk.*
+import skunk.codec.boolean.bool
+import skunk.implicits.*
+
+import java.time.Duration
+
+import Services.Syntax.*
+
+sealed trait ObservationWorkflowService[F[_]] {
+
+  def observationValidations(
+    oid: Observation.Id,
+    itcClient: ItcClient[F],
+  )(using Transaction[F]): F[Result[List[ObservationValidation]]]
+
+  def getWorkflow(
+    oid: Observation.Id, 
+    commitHash: CommitHash, 
+    itcClient: ItcClient[F]
+  )(using Transaction[F]): F[Result[ObservationWorkflow]]
```

Comment:
This new service method is unused. It will be the entry point for computing the workflow value once it's in the schema.